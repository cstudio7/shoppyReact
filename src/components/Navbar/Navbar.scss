.mainContainer {
  height: 100%;
}
.pageContent {
  margin-top: 64px;
}

@media (min-width: 769px) {
  .pageContent {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
  }
}
