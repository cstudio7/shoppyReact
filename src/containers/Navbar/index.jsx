import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NavbarComponent from '../../components/Navbar';

export default class Navbar extends Component {
  render() {
    return <NavbarComponent userRole="admin" displayPage={this.props.displayPage} />;
  }
}
Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};
