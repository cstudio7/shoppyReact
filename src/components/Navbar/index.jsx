import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Toolbar from '../Toolbar/Toolbar';
import SideDrawer from '../SideDrawer/SideDrawer';
import Backdrop from '../Backdrop/Backdrop';
import './Navbar.scss';

class Navbar extends Component {
  state = {
    sideDrawerOpen: false
  };

  drawerToggleClickHandler = () => {
    this.setState(prevState => ({ sideDrawerOpen: !prevState.sideDrawerOpen }));
  };

  backdropClickHandler = () => {
    this.setState({ sideDrawerOpen: false });
  };

  render() {
    let backdrop;

    if (this.state.sideDrawerOpen) {
      backdrop = <Backdrop click={this.backdropClickHandler} />;
    }
    return (
      <div className="mainContainer">
        <Toolbar drawerClickHandler={this.drawerToggleClickHandler} />
        <SideDrawer show={this.state.sideDrawerOpen} />
        {backdrop}
        <main className="pageContent">{this.props.displayPage}</main>
      </div>
    );
  }
}

Navbar.propTypes = {
  displayPage: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
};

export default Navbar;
