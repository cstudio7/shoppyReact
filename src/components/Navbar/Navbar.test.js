import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Navbar from './index';
import Component from '../AuthForm';

describe('Test the Navbar Component', () => {
  it('should match snapshot', () => {
    const click = jest.fn();
    const wrapper = shallow(<Navbar displayPage={Component} click={click} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().sideDrawerOpen).toEqual(false);
    wrapper.instance().drawerToggleClickHandler();
    wrapper.instance().backdropClickHandler();
  });
});
