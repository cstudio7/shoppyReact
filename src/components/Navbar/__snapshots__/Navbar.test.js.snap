// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Navbar Component should match snapshot 1`] = `
<div
  className="mainContainer"
>
  <toolbar
    drawerClickHandler={[Function]}
    userRole="attendant"
  />
  <SideDrawer
    show={false}
    userRole="attendant"
  />
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={true}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconId={null}
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={false}
    modalId={null}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    overlayId={null}
    showCloseIcon={true}
    styles={Object {}}
  />
  <main
    className="pageContent"
  >
    <Component />
  </main>
</div>
`;
